packages/expo-asset/tools/hashAssetFiles.js

```diff
@@ -15,6 +15,19 @@ module.exports = function hashAssetFiles(asset) {
         .replace(/\.\.\//g, '_');
     }
 
+    // URL encode asset paths defined as `?export_path` or `?unstable_path` query parameters.
+    // Decoding should be done automatically when parsing the URL through Node or the browser.
+    const assetPathQueryParameter = asset.httpServerLocation.match(
+      /\?(export_path|unstable_path)=(.*)/
+    );
+    if (assetPathQueryParameter && assetPathQueryParameter[2]) {
```

Comment:
nit: this could look a bit better with 👇 
```suggestion
    if (assetPathQueryParameter?.[2]) {
```

Reply:
Yeah, I wasn't sure if we could use that since this file is _NOT_ being transpiled. It's just a plain JS file being shipped as is, so I also avoided constructor magic 🙈